webpush/encrypt_test.go

```diff
@@ -202,6 +231,37 @@ func TestRfcVectors(t *testing.T) {
 	}
 }
 
+// TestAES128gcmRfcVectors uses the values given in the RFC for HTTP encryption to verify
+// that the code conforms to the RFC
+// See: https://tools.ietf.org/html/draft-ietf-webpush-encryption-07#appendix-A
+func TestAES128gcmRfcVectors(t *testing.T) {
+	defer stubFuncs(rfcAES128gcmSalt, rfcAES128gcmKeys)()
+	b64 := base64.URLEncoding.WithPadding(base64.NoPadding)
+
+	auth, err := b64.DecodeString(rfcAES128gcmAuth)
+	if err != nil {
+		t.Error(err)
+	}
+	key, err := b64.DecodeString(rfcAES128gcmPublic)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sub := &Subscription{Auth: auth, Key: key}
+	result, err := Encrypt(sub, aes128gcmMessage, AES128GCM)
+	if err != nil {
+		t.Error(err)
```

Comment:
`t.Fatal` here and everywhere else, where the return value is being used. You can't rely on the `result` below if err is returned.